optimizely/optimizely.py

```diff
@@ -629,8 +632,11 @@ def get_variation(
             return None
 
         user_context = OptimizelyUserContext(self, self.logger, user_id, attributes, False)
-
-        variation, _ = self.decision_service.get_variation(project_config, experiment, user_context)
+        user_profile_tracker = user_profile.UserProfileTracker(user_id, self.user_profile_service, self.logger)
+        variation, _ = self.decision_service.get_variation(project_config,
+                                                           experiment,
+                                                           user_context,
+                                                           user_profile_tracker)
```

Comment:
the user profile should be saved here as we will remove the save from get_variation() in decision service.